wave_geometry/include/wave/geometry/exception_helpers.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef WAVE_EXCEPTION_HELPERS_HPP
+#define WAVE_EXCEPTION_HELPERS_HPP
+
+#include "wave/geometry/rotation.hpp"
+#include "wave/utils/math.hpp"
+
+namespace wave {
+
+
+// This function checks to make sure all elements of the input
+// Eigen array have finite values (not Nan or +/- Inf), as these
+// values are invalid for rotation geometry.
+template <typename MatrixType>
+void checkArrayFinite(const Eigen::MatrixBase<MatrixType> &mat) {
```

Comment:
Minor naming thing, but I don't know if this should be named `checkMatVecFinite` instead of array? or even specific like `checkVecFinite` and `checkMatFinite`? last thing do you think maybe C style functions like this should be snake case (i.e. all lower, separated by underscores: `this_is_a_snake`)? to be able to quickly differentiate between class methods vs functions. Food for thought.